### PR TITLE
HDDS-8456. Support read-only administrators in SCM

### PR DESCRIPTION
**Pull request by z-bb**
## What changes were proposed in this pull request?

Scm supports read only administrators.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8456

## How was this patch tested?

ut

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java [hunk 1]
```diff
@@ -1876,16 +1879,31 @@ public boolean checkLeader() {
     }
   }
 
+  public void setScmReadOnlyAdmins(OzoneAdmins scmReadOnlyAdmins) {
+    this.scmReadOnlyAdmins = scmReadOnlyAdmins;
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java:1883]: Would be nice to omit this, the test can set configuration via `OzoneConfiguration`.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java [hunk 2]
```diff
@@ -1876,16 +1879,31 @@ public boolean checkLeader() {
...
+    switch (op) {
+      case "startReconfiguration":
+        checkAdminAccess(getRemoteUser(), false);
+        break;
+      case "getReconfigurationStatus":
+      case "listReconfigurableProperties":
+        checkAdminAccess(getRemoteUser(), true);
+        break;
+      default:
+    }
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java:1896]: `default` branch should check write admin access.

**Review by adoroszlai — COMMENTED** [review]
Thanks @z-bb for working on this.

@z-bb please check failures:

https://github.com/z-bb/ozone/actions/runs/5309440423/jobs/9610104266#step:6:528

and 

https://github.com/z-bb/ozone/actions/runs/5309440423/jobs/9610548522#step:5:280

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java (hunk 2 above)
Comment (z-bb) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java]: @adoroszlai Thanks for review, I made some changes, please review again.

**Comment by adoroszlai** [comment]
Thanks @z-bb for updating the patch.  The same checks are still failing:

https://github.com/z-bb/ozone/actions/runs/5319769143/jobs/9632667900
https://github.com/z-bb/ozone/actions/runs/5319769143/jobs/9633149405

**Comment by z-bb** [comment]
> Thanks @z-bb for updating the patch. The same checks are still failing:
> 
> https://github.com/z-bb/ozone/actions/runs/5319769143/jobs/9632667900 https://github.com/z-bb/ozone/actions/runs/5319769143/jobs/9633149405

@adoroszlai Thanks for review, I redid the next modification

**Review by adoroszlai — APPROVED** [review]
Thanks @z-bb for updating the patch, LGTM.

**Pull request merged**